File: app/controllers/BlanksTestController.java
package controllers;

import controllers.core.BaseController;
import model.blanks.BlanksExercise;
import model.result.SuccessType;
import play.data.DynamicForm;
import play.data.FormFactory;
import play.libs.Json;
import play.mvc.Result;
import play.mvc.Security;

import javax.inject.Inject;
import java.util.ArrayList;
import java.util.List;

@Security.Authenticated(model.Secured.class) public class BlanksTestController extends BaseController {

  private static final BlanksExercise exercise = new BlanksExercise(1);

  @Inject public BlanksTestController(FormFactory f) {
    super(f);
  }

  public Result correctBlanks(int id) {
    DynamicForm form = factory().form().bindFromRequest();
    int inputCount = Integer.parseInt(form.get("count"));

    List<String> inputs = new ArrayList<>(inputCount);
    for(int count = 0; count < inputCount; count++)
      inputs.add(form.get("inp" + count));

    List<SuccessType> results = exercise.correct(inputs);

    return ok(Json.toJson(results));
  }

  public Result testBlanks() {
    return ok(views.html.blanks.render(getUser(), exercise));
  }
}

File: app/controllers/BlanksTestController.scala
package controllers

import javax.inject.Inject

import controllers.core.BaseController
import model.blanks.BlanksExercise
import play.api.mvc.ControllerComponents
import play.mvc.Security

@Security.Authenticated(classOf[model.Secured])
class BlanksTestController @Inject()(cc: ControllerComponents) extends BaseController(cc) {

  private val exercise = new BlanksExercise


  def correctBlanks(id: Int) = Action { implicit request =>
    //    val form = factory.form().bindFromRequest()
    //    val inputCount = Integer.parseInt(form.get("count"))
    //
    //    val inputs = (for (count <- 0 until inputCount) yield form.get(s"inp$count")).toList
    //
    //    ok(Json.toJson(exercise.correct(inputs)))
    Ok("TODO")
  }

  def testBlanks = Action { implicit request => Ok(views.html.blanks.render(getUser, exercise)) }
}
